app/models/concerns/profile_searching.rb

```diff
@@ -148,6 +154,24 @@ def filter_by_policy(_filter, _operator, policy_or_profile_id)
         conditions: "profiles.id IN (#{any_child_profile.to_sql})"
       }
     end
+
+    def filter_by_policy_and_profile_names(_filter, operator, value)
+      ss_qb = ScopedSearch::QueryBuilder.new(scoped_search_definition, '', scoped_search_definition.profile)
+      # Passing empty block because nothing needs to happen on yield in preprocess_parameters
+      # The correct yield is invoked in to_ext_method_sql so all notification values are updated correctly
+      # See https://github.com/wvanbergen/scoped_search/blob/master/lib/scoped_search/query_builder.rb
+      values = ss_qb.preprocess_parameters(
+        scoped_search_definition.field_by_name('name'), OPERATORS[operator], value
+      ) { |x| }
+
+      policy_cond = sanitize_sql_for_conditions(["policies.name #{operator} (?)", values])
+      profile_cond = sanitize_sql_for_conditions(["profiles.name #{operator} (?)", values])
```

Comment:
Why are these arrays?

Reply:
I thought that [sanitize_sql_for_conditions](https://api.rubyonrails.org/classes/ActiveRecord/Sanitization/ClassMethods.html#method-i-sanitize_sql_for_conditions) only accepted arrays but I just checked and it also accepts strings. I'll update it to a string!

Reply:
Actually, it does work better passing the array into the [sanitize_sql_for_conditions](https://api.rubyonrails.org/classes/ActiveRecord/Sanitization/ClassMethods.html#method-i-sanitize_sql_for_conditions) call since it takes the values parameter. `sanitize_sql_for_conditions` takes an array or a string.